src/foam/nanos/auth/UserAndGroupAuthService.js

```diff
@@ -0,0 +1,609 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'UserAndGroupAuthService',
+  flags: ['java'],
+
+  implements: [
+    'foam.nanos.NanoService',
+    'foam.nanos.auth.AuthService'
+  ],
+
+  imports: [
+    'localUserDAO',
+    'localGroupDAO',
+    'localSessionDAO'
+  ],
+
+  javaImports: [
+    'foam.core.ContextAwareSupport',
+    'foam.core.X',
+    'foam.dao.ArraySink',
+    'foam.dao.DAO',
+    'foam.dao.Sink',
+    'foam.mlang.MLang',
+    'foam.nanos.logger.Logger',
+    'foam.nanos.NanoService',
+    'foam.nanos.session.Session',
+    'foam.util.Email',
+    'foam.util.Password',
+    'foam.util.SafetyUtil',
+
+    'javax.security.auth.AuthPermission',
+    'java.security.Permission',
+    'java.util.Calendar',
+    'java.util.List',
+    'java.util.regex.Pattern',
+
+    'static foam.mlang.MLang.AND',
+    'static foam.mlang.MLang.EQ'
+  ],
+
+  axioms: [
+    {
+      name: 'javaExtras',
+      buildJavaClass: function(cls) {
+        cls.extras.push(`
+          public final static String CHECK_USER_PERMISSION = "service.auth.checkUser";
+        `);
+      }
+    }
+  ],
+
+  constants: [
+    {
+      name: 'PASSWORD_VALIDATE_REGEX',
+      type: 'String',
+      value: '^.{6,}$'
+    },
+    {
+      name: 'PASSWORD_VALIDATION_ERROR_MESSAGE',
+      type: 'String',
+      value: 'Password must be at least 6 characters long.'
+    }
+  ],
+
+  properties: [
+    {
+      class: 'foam.dao.DAOProperty',
+      name: 'userDAO',
+      javaFactory: 'return (DAO) getLocalUserDAO();'
+    },
+    {
+      class: 'foam.dao.DAOProperty',
+      name: 'groupDAO',
+      javaFactory: 'return (DAO) getLocalGroupDAO();'
+    },
+    {
+      class: 'foam.dao.DAOProperty',
+      name: 'sessionDAO',
+      javaFactory: 'return (DAO) getLocalSessionDAO();'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'start',
+      javaCode: `
+        getUserDAO();
```

Comment:
Why do you need to touch all of these on start?

Reply:
I don't- will remove.